### ISSUE TITLE
Ongoing crashes after upgrade

### ISSUE THREAD

**Issue by fah** [atom/atom#15443]
Atom is not usable anymore.
After an upgrade a few minutes ago it crashes every 2-5 minutes.

------------------------------

Process:               Atom Helper [16341]
Path:                  /Applications/Atom.app/Contents/Frameworks/Atom Helper.app/Contents/MacOS/Atom Helper
Identifier:            com.github.atom.helper
Version:               1.19.3 (1.19.3)
Code Type:             X86-64 (Native)
Parent Process:        Atom [15582]
Responsible:           Atom [15582]
User ID:               501

Date/Time:             2017-08-25 18:29:00
OS Version:            Mac OS X 10.12.6 (16G29)
Report Version:        12

Time Awake Since Boot: 100000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrRendererMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000

**Comment by maxbrunsfeld** [comment]
Is there more to the crash log? Usually there are stack traces for each thread in the crash report in `Console.app`. What were you doing when Atom crashed?

**Comment by maxbrunsfeld** [comment]
Also, does it happen if you run Atom in safe mode (`atom --safe`)? If not, could you report the list of packages you have installed?

**Comment by fah** [comment]
Since I could not work on I replaced the Atom.app with a backup of before the upgrade happened.
So I can't test --safe anymore.

One of the logs is attached. They look all the same at the first glance.

[Atom Helper_2017-08-25-182802_iMac.crash.txt](https://github.com/atom/atom/files/1253529/Atom.Helper_2017-08-25-182802_iMac.crash.txt)

**Comment by fah** [comment]
Atom crashed by its own. It didn't not matter what I did. Also without any open window. After a delay It crashed.

**Comment by fah** [comment]
Packages: Since there are 17 (Community) + 78 (Core) + 1 (Git) Packages installed I would prefer a copy and paste solution to show you the list. Any hint how to do that?

**Comment by maxbrunsfeld** [comment]
`apm list --installed`

**Comment by fah** [comment]
$ apm list --installed
Community Packages (17) /Users/f/.atom/packages
├── autocomplete-clang@0.11.3
├── build@0.68.0
├── busy@0.7.0
├── file-icons@2.1.11
├── git-control@0.9.0
├── git-plus@7.9.3
├── git-time-machine@1.5.9
├── intentions@1.1.5
├── language-ini@1.19.0
├── linter@2.2.0
├── linter-gcc@0.7.1
├── linter-ui-default@1.6.7
├── minimap@4.29.6
├── plantuml-viewer@0.7.2
├── platformio-ide-debugger@1.2.4
├── platformio-ide-terminal@2.6.0
└── tool-bar@1.1.0

Git Packages (1) /Users/f/.atom/packages
└── platformio-ide@2.0.0-beta.8 (platformio/platformio-atom-ide#fda347eb)

**Comment by SheikhYourass** [comment]
I'm having exactly the same problem: crashes every 2 to 5 minutes, even when atom is just opened and I'm not working. Also writing and scrolling gets interrupted for about a second every 5-10 secs or so. Same issues in safe mode.

Community Packages (14) 
├── Remote-FTP@1.1.3
├── busy-signal@1.4.3
├── color-picker@2.2.5
├── file-icons@2.1.11
├── git-plus@7.9.3 (disabled)
├── intentions@1.1.5
├── linter@2.2.0
├── linter-sass-lint@1.7.4
├── linter-scss-lint@3.1.1 (disabled)
├── linter-ui-default@1.6.8
├── minimap@4.29.6 (disabled)
├── minimap-pigments@0.2.2 (disabled)
├── pigments@0.40.2
└── sass-autocompile@0.13.3

Crash log
[Atom Helper_2017-08-29-155147_Johannes-MacBook.crash.zip](https://github.com/atom/atom/files/1260518/Atom.Helper_2017-08-29-155147_Johannes-MacBook.crash.zip)

**Comment by spcat1212** [comment]
I've begun experiencing the same issues. I've disabled all Custom packages and only have Core packages running. Getting the same results while running `atom --safe`.


[Atom Helper_2017-08-09-121902_stevecat.crash.zip](https://github.com/atom/atom/files/1263897/Atom.Helper_2017-08-09-121902_stevecat.crash.zip)

**Comment by maxbrunsfeld** [comment]
Do the crashes go away if you disable the built-in `github` package? You can do `apm disable github` at the command line.

**Comment by SheikhYourass** [comment]
Wow, yes! No more crashes. Perfect, thanks Max!

**Comment by maxbrunsfeld** [comment]
@SheikhYourass Ok, I think the issue you're experiencing is being tracked here: https://github.com/atom/github/issues/953. Really sorry for the poor experience. You can subscribe to that issue for updates on this.

**Comment by BinaryMuse** [comment]
@maxbrunsfeld The exception type is the same as the one in atom/github#953, but the stack is different. Previously we were consistently getting into `v8::internal::V8::FatalProcessOutOfMemory`, presumably from the porcelain status parser using too much memory.

Here, we seem to be seeing a crash in buffer-related code related to IPC event handling (`atom::AtomRenderViewObserver::EmitIPCEvent`). I *am* wondering if it's still related to the large amount of output from the `status --porcelain=v2` call...

**Comment by maxbrunsfeld** [comment]
Yeah, I was just referencing that issue because it's a crash, it's open, and we've identified that this crash is related to the github package. Maybe it makes sense to reopen a separate issue for this one.

**Comment by BinaryMuse** [comment]
Okay, I'm able to reproduce this by adding a Git repo at `~/.git` and opening my home directory in Atom. Interestingly, there seems to be a large amount of synchronous work blocking the UI thread associated with IPC events, *presumably* from sending the output of the status command back to the primary renderer process. I wonder if switching to web workers would fix this (cc @kuychaco); I also wonder if everyone experiencing this issue has a Git repo in their home directory.

@SheikhYourass, @spcat1212, and @fah — if you're running a relatively recent version of Git, would you mind running the following on the Terminal and telling me what the output is? It may take just a moment to run:

```
GIT_DIR="$HOME/.git" GIT_WORK_TREE="$HOME" git status --porcelain=v2 --branch --untracked-files=all --ignore-submodules=dirty | wc -l
```

Happy to move this issue to `atom/github`, but there's already a lot of context here so I'm tempted to keep it for now.

**Comment by deepmehtait** [comment]
My Atom is not opening after crash. Getting this error as shown in screenshot below.
<img width="987" alt="screen shot 2017-09-01 at 4 54 29 pm" src="https://user-images.githubusercontent.com/8682058/29991095-97be8dae-8f36-11e7-8c4f-f42121d8c7b3.png">

I tried to install new version as well but still getting the same issue.

**Comment by unforswearing** [comment]
Also experiencing this issue, though it seems to happen when switching window focus to Atom from Google Chrome (v. 60.0.3112.113). `apm disable github` and `atom --safe` did not work for me. 

Community Packages (14)
├── atom-html-preview@0.1.22
├── auto-update-packages@1.0.1
├── busy-signal@1.4.3
├── github-atom-light-syntax@0.5.0
├── intentions@1.1.5
├── language-toml@0.19.0 (disabled)
├── linter@2.2.0
├── linter-shellcheck@1.4.6
├── linter-ui-default@1.6.8
├── minimap@4.29.6
├── polymorph-ui@1.8.8
├── seafoam-pastel-syntax@1.1.0
├── soft-muted@0.3.0
└── zenburn@3.1.0

Crash Log
[atom.log.zip](https://github.com/atom/atom/files/1271845/atom.log.zip)
